scripts/run_commands.sh

```diff
@@ -0,0 +1,92 @@
+#
+# This scripts runs on the ec2 instance. It is uploaded to the EC2 instance by
+# test_commands.sh
+# The necessary env variables will be set by test_commands.sh using the
+# command line arguments provided to it. 
+#
+
+declare -i FAILURES
+FAILURES=0
+
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+LRED='\033[0;31m'
+LGREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+expect_success() {
+	if "${@}" &>> ~/test_output.txt; then
+		echo -e "${GREEN}SUCCEEDED${NC}: ${LGREEN}${@}${NC}"
+	else
+		echo "-------"
+		echo -e "${RED}FAILED${NC}: ${LRED}${@}${NC}"
+		echo "-------"
+		FAILURES+=1
+	fi
+}
+
+# used for commands that whose arguments might contain sensitive information
+expect_success_no_log() {
+	if "${@}" &>> ~/test_output.txt; then
+		echo -e "${GREEN}SUCCEEDED${NC}: ${LGREEN}${1} ${2}${NC}"
+	else
+		echo "-------"
+		echo -e "${RED}FAILED${NC}: ${LRED}${1} ${2}${NC}"
+		echo "-------"
+		FAILURES+=1
+	fi
+}
+
+# install git and go
+echo "TYPE y|yes then enter to proceed."
```

Comment:
Can we move the comments here?

"Please make sure you have git and go installed. Do you wish to continue? [y/n]"

Reply:
This runs on the instance, which is presumably a clean instance without any of those installed on it. 

I don't understand what you mean be "Can we move the comments here"

Reply:
If I didn't read the script, I would not know by proceeding, go and git will be installed.

I thought the script assumes you already have an EC2 instance, it doesn't spin up an EC2 instance for you. I can use the same instance to run the integration test multiple times. Is it really necessary to install go and git on the EC2 instance? Why can't we just copy the binary needed for testing? With the setup today, I must check in my code and must of a forked repository in order for this to work.

Reply:
> Why can't we just copy the binary needed for testing?

I'm looking into it.

> With the setup today, I must check in my code and must of a forked repository in order for this to work.

Actually it can also test branches on the main repo, by setting the GitHub username to be "aws".

(I do understand that the script is not ideal as it functions now)